doc.go

```diff
@@ -22,8 +22,183 @@ instance:
+matches the provided regular expression pattern.
+For instance:
+
+    c.Assert("these are the voyages", qt.Matches, "these are .*")
+    c.Assert(net.ParseIP("1.2.3.4"), qt.Matches, "1.*")
+
+ErrorMatches
+
+ErrorMatches is a Checker checking that the provided value is an error whose
+message matches the provided regular expression pattern.
+For instance:
+
+    c.Assert(err, qt.ErrorMatches, "bad wolf .*")
+
+PanicMatches
+
+PanicMatches is a Checker checking that the provided function panics with a
+message matching the provided regular expression pattern.
+For instance:
+
+    c.Assert(func() {panic("bad wolf ...")}, qt.PanicMatches, "bad wolf .*")
+
+IsNil
+
+IsNil is a Checker checking that the provided value is nil.
+For instance:
+
+    c.Assert(got, qt.IsNil)
+
+HasLen
+
+HasLen is a Checker checking that the provided value has the given length.
+For instance:
+
+    c.Assert([]int{42, 47}, qt.HasLen, 2)
+    c.Assert(myMap, qt.HasLen, 42)
+
+Satisfies
+
+Satisfies is a Checker checking that the provided value, when used as
+argument of the provided predicate function, causes the function to return
+true. The function must be of type func(T) bool, having got assignable to T.
+For instance:
+
+    // Check that an error from os.Open satisfies os.IsNotExist.
+    c.Assert(err, qt.Satisfies, os.IsNotExist)
+
+    // Check that a floating point number is a not-a-number.
+    c.Assert(f, qt.Satisfies, math.IsNaN)
+
+Not
+
+Not returns a Checker negating the given Checker.
+For instance:
+
+    c.Assert(got, qt.Not(qt.IsNil))
+    c.Assert(answer, qt.Not(qt.Equals), 42)
+
+Contains
+
+Contains is a checker that checks that a map, slice, array
+or string contains a value. It's the same as using
+Any(Equals), except that it has a special case
+for strings - if the first argument is a string,
+the second argument must also be a string
+and strings.Contains will be used.
+
+For example:
+
+	c.Assert("hello world", qt.Contains, "world")
+	c.Assert([]int{3,5,7,99}, qt.Contains, 7)
+
+Any
+
+Any returns a Checker that uses the given checker to check elements
+of a slice or array or the values from a map. It succeeds if any element
+passes the check.
+
+For example:
+
+	c.Assert([]int{3,5,7,99}, qt.Any(qt.Equals), 7)
+	c.Assert([][]string{{"a", "b"}, {"c", "d"}}, qt.Any(qt.DeepEquals), []string{"c", "d"})
+
+See also All and Contains.
+
+All
+
+All returns a Checker that uses the given checker to check elements
+of slice or array or the values of a map. It succeeds if all elements
+pass the check.
+On failure it prints the error from the first index that failed.
+
+For example:
+
+	c.Assert([]int{3, 5, 8}, qt.All(qt.Not(qt.Equals)), 0)
+	c.Assert([][]string{{"a", "b"}, {"a", "b"}}, qt.All(qt.DeepEquals), []string{"c", "d"})
+
+See also Any and Contains.
+
+Deferred execution
+
+Quicktest provides the ability to defer the execution of functions that will be
+run in last-in, first-out order when the test completes. This is often useful
```

Comment:
Maybe keep the "last-in, first-out" thing for a later sentence?

```
Quicktest provides the ability to defer the execution of functions that will be
run when the test completes. This is often useful.... etc.
The functions will be run in last-in, first-out order.
```

?

Reply:
Done.